theano/compile/function_module.py

```diff
@@ -676,6 +679,17 @@ def __call__(self, *args, **kwargs):
         elif self.unpack_single and len(outputs) == 1:
             return outputs[0]
         else:
+
+            if self.output_dictionary_flag: 
+                outputDict = {}
+
+                assert len(self.output_keys) == len(outputs)
+
+                for i in range(0, len(self.output_keys)): 
+                    outputDict[self.output_keys[i]] = outputs[i]
+
+                outputs = outputDict
```

Comment:
I was just wondering if maybe you could have simplified things a bit by just storing the dict in ``outputs`` instead if adding a new argument. But I had a quick look and I'm not sure it would actually be simpler since there's a bunch of code relying on the fact ``outputs`` is a list. So forget about it :)

Just two small code style points to fix:
1. Use ``is not None`` instead of ``!= None``
2. Your ``for`` loop can be written as: ``outputDict = dict(itertools.izip(self.output_keys, outputs))`` (actually you can just do ``return dict(...)`` without going through intermediary variables ``outputDict`` and ``outputs``)